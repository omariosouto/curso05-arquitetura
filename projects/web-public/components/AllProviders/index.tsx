import { ThemeProvider } from "styled-components";

const theme = {
  colors: {
    primary: "#0070f3",
  },
};

export function AllProviders({ children }) {
  return (
    <ThemeProvider theme={theme}>
      Provider Here!
      {children}
    </ThemeProvider>
  );
}
