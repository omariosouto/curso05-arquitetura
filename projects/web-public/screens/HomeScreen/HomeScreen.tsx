import { Text } from "@alura/design-system";
import { sum } from "@alura/utils/math/sum";

export function HomeScreen() {
  return (
    <div>
      <Text tag="h1">Home do Projeto</Text>
      <Text tag="p">@alura/utils/math/sum: sum(2,2) = {sum(2, 2)}</Text>
    </div>
  );
}
