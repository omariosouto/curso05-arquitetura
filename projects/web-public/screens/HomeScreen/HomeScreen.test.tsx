import { customRender } from "@alura/test-commons/react-testing-library";
import { AllProviders } from "../../components/AllProviders";
import { HomeScreen } from "./HomeScreen";

// Aqui podemos customizar os valores padrão dos providers para cada teste
const render = customRender((props) => <AllProviders {...props} />);

describe("<HomeScreen />", () => {
  it("renders the page", () => {
    const { container } = render(<HomeScreen />);
    // Para páginas provavemlente não iremos fazer teste de snapshot, mas fica o demo
    expect(container).toMatchSnapshot();
  });
});
