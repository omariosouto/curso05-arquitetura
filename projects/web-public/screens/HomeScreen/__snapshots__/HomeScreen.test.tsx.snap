// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders the page 1`] = `
.c0 {
  font-family: sans-serif;
}

<div>
  Provider Here!
  <div>
    <h1
      class="c0"
    >
      Home do Projeto
    </h1>
    <p
      class="c0"
    >
      @alura/utils/math/sum: sum(2,2) = 
      4
    </p>
  </div>
</div>
`;
